primefaces/src/main/java/org/primefaces/util/ComponentUtils.java

```diff
@@ -632,7 +637,22 @@ public static void encodeIndexedId(FacesContext context, UIComponent component,
         // append index to all id's
         char separator = UINamingContainer.getSeparatorChar(context);
         String encodedComponent = fsw.toString();
-        encodedComponent = encodedComponent.replaceAll("\\sid=\"(.*?)\"", " id=\"$1" + separator + index + "\"");
+
+        // find all id's to replace
+        Matcher matcher = ID_PATTERN.matcher(encodedComponent);
+
+        // grab all the unique ID's found
+        Set<String> ids = new HashSet<>(5);
+        while (matcher.find()) {
+            ids.add(matcher.group(1));
+        }
+
+        // replace each id with an indexed version
+        for (String id : ids) {
+            String replaceId = id + separator + index;
+            encodedComponent = encodedComponent.replaceAll(id, replaceId);
+        }
```

Comment:
You could use one loop instead of two no?

Reply:
I was thinking that wasn't suure if manipulating the underlying string would affect the matcher while its in progress.  Let me try it.

Reply:
1 loop works!